lib/active_shipping/carriers/ups.rb

```diff
@@ -800,6 +821,17 @@ def parse_delivery_dates_response(origin, destination, packages, response, optio
       response = DeliveryDateEstimatesResponse.new(success, message, Hash.from_xml(response).values.first, :delivery_estimates => delivery_estimates, :xml => response, :request => last_request)
     end
 
+    def parse_void_response(response, options={})
+      xml = build_document(response, 'VoidShipmentResponse')
+      success = response_success?(xml)
+      message = response_message(xml)
+      if success
+        true
+      else
+        raise "Void shipment failed with message: #{message}"
```

Comment:
Either return a falsy response object, or raise a typed exception. You'll want to be able to catch it and perform an action. Raising a standard error makes it difficult to cleanly recover from.